datayoga-py/datayoga_core/resources/scaffold/jobs/sample/hello.yaml

```diff
@@ -1,9 +1,19 @@
 input:
-  uses: std.read
+  uses: files.read_csv
+  with:
+    file: sample.csv
 steps:
   - uses: add_field
     with:
-      field: greeting
-      expression: "'hello ' || name"
-      language: sql
+      fields:
+        - field: full_name
+          language: jmespath
+          expression: concat([capitalize(fname), ' ' , capitalize(lname)])
+        - field: greeting
+          language: sql
+          expression: "'Hello ' || CASE WHEN gender = 'F' THEN 'Ms.' WHEN gender = 'M' THEN 'Mr.' ELSE 'N/A' END || ' ' || full_name"
```

Comment:
this is interesting. reusing a field within a field. this is what Informatica does. We should probably add a test for this.

Reply:
why does it matter if this `full_name` field was defined here or in a previous block?